hardware/hardware.sh

```diff
@@ -130,8 +130,20 @@ else
 fi
 echo
 
-echo "Instance type from IMDS (if available):"
-curl -s --connect-timeout 1 'http://169.254.169.254/latest/meta-data/instance-type' | tee instance.txt
+echo "Instance type from Metadata service (if available):"
+AWS_HTTP_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 'http://169.254.169.254/latest/meta-data/')
+if [[ "$AWS_HTTP_CODE" == "200" ]]; then
+    AWS_INSTANCE_TYPE=$(curl -s --connect-timeout 1 'http://169.254.169.254/latest/meta-data/instance-type')
+    echo "$AWS_INSTANCE_TYPE" > instance.txt
+    exit 0
```

Comment:
So here and in l. 143, we are in the happy path. I don't quite understand why we exit the script at this point (l. 137/145)?

Reply:
You are right, that snuck in from when I was testing in isolation. Fixed.